build.gradle

```diff
@@ -85,8 +85,31 @@ if (JavaVersion.current() == JavaVersion.VERSION_1_8) {
                               "--add-opens", "java.base/sun.nio.cs=ALL-UNNAMED",
                               "--add-opens", "java.base/sun.security.action=ALL-UNNAMED",
                               "--add-opens", "java.base/sun.util.calendar=ALL-UNNAMED"]
+} else if (JavaVersion.current() == JavaVersion.VERSION_21) {
+  project.ext.jdkVersion = '21'
+  project.ext.extraJvmArgs = ["--add-opens", "java.base/java.io=ALL-UNNAMED",
```

Comment:
extra JVM args seem to be the same as needed for JDK17, so maybe just add a `JavaVersion.current() >= JavaVersion.VERSION_17` and set `project.ext.jdkVersion` via `JavaVersion.current().getMajorVersion()`

Reply:
yeah that's why I mentioned to set `project.ext.jdkVersion` via `JavaVersion.current().getMajorVersion()` for JDK17+

Reply:
missed that, thanks! updated